src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +223,437 @@ def _updateImage(self):
+                    # Does not matter from which of the 6 images as they all must have been recorded with the same
+                    # filter -> reuse key  # TODO dont like this solution, is there a better one?
+                    fallback_wl = 650e-9
+                    band = data_raw_bg_corr[key].metadata.get(model.MD_OUT_WL, fallback_wl)
+                    if isinstance(band, tuple):  # wl is usually tuple of min/max value
+                        wl = sum(band) / len(band)
+                    else:  # handles if band is str TODO if "str", support center wavelength based on color
+                        wl = fallback_wl
+
+                    # Warning: allocates lot of memory, which will not be free'd until
+                    # the current thread is terminated.
+
+                    # TODO allow variable input size? Calc based on raw data?
+                    size = (180, 720)  # (2*90, 2*360) defines the resolution of the displayed image
+
+                    for key in data_raw_bg_corr.keys():
+                        data_raw_bg_corr[key] = angleres.AngleResolved2Rectangular(data_raw_bg_corr[key], size)
+                    polarimetry_dict_raw[ebeam_pos] = data_raw_bg_corr  # dict containing the bg corrected raw data
+
+                    # Calculate the polarimetry results:
+                    # Dictionary keys are (ebeam pos x, ebeam pos y, optional: polarimetry key).
+                    # Note: Takes about 0.25 sec to calc all polarimetry results for one ebeam pos
+                    # and tested on an image of size (256, 1024)
+                    polarimetry_dict_raw.update(
+                        arpolvis.calcPolarimetry(polarimetry_dict_raw, ebeam_pos, size, wl))
+
+                # Convert the data back to polar representation for GUI display.
+                # Note: Takes 0.24 sec to convert one image for display and tested on an image of size (256, 1024)
+                # TODO invalid division check if only bg  # norm so far
+                # TODO move to update image?? as this is the only code we need to reuse
+                #  whenever the legend changes while having the same ebeam pos selected
+                if pos[-1] in ["Ephi", "Etheta", "Ex", "Ey", "Ez"]:  # TODO: hardcoded for now
+                    data = numpy.abs(polarimetry_dict_raw[pos])
+                    order = numpy.log10(data.max())
+                    try:
+                        plotorder = numpy.int(order)
+                    except OverflowError:
+                        # Note: Error is raised, when bg image and image are the same
+                        # (-> cannot convert float infinity to integer)
+                        # TODO is there a better way to handle this problem?
+                        plotorder = 0
+                    polarimetry_dict_raw[pos] = data / 10 ** plotorder
+                    colormap = 2
```

Comment:
Please comment what the colormap 2 and colormap 1 are. If the Rectangular2Polar was taking a meaningful string or a constant (eg CMAP_EELS) instead of a number, that could be avoided.